nestkernel/simulation_manager.cpp

```diff
@@ -435,28 +499,46 @@ nest::SimulationManager::simulate( Time const& t )
   // above.
   if ( t.get_steps() % kernel().connection_manager.get_min_delay() != 0 )
     LOG( M_WARNING,
-      "SimulationManager::simulate",
+      "SimulationManager::run",
       "The requested simulation time is not an integer multiple of the minimal "
       "delay in the network. This may result in inconsistent results under the "
       "following conditions: (i) A network contains more than one source of "
       "randomness, e.g., two different poisson_generators, and (ii) Simulate "
       "is called repeatedly with simulation times that are not multiples of "
       "the minimal delay." );
 
-  resume_( num_active_nodes );
+  start_updating_();
+}
+
+void
+nest::SimulationManager::cleanup()
+{
+  if ( not simulated_ )
+    return;
 
-  finalize_simulation_();
+  // Check for synchronicity of global rngs over processes
+  if ( kernel().mpi_manager.get_num_processes() > 1 )
+    if ( !kernel().mpi_manager.grng_synchrony(
+           kernel().rng_manager.get_grng()->ulrand( 100000 ) ) )
+    {
+      throw KernelException(
+        "In SimulationManager::cleanup(): "
+        "Global Random Number Generators are not "
+        "in sync at end of simulation." );
+    }
+
+  kernel().node_manager.finalize_nodes();
 }
 
 void
-nest::SimulationManager::resume_( size_t num_active_nodes )
+nest::SimulationManager::start_updating_()
```

Comment:
I am not really happy with the name `start_updating_()` because it makes me look for `stop_updating_()`. Since we already have `update_()`, maybe `update_network_()` or `perform_update_ ()` would be better.